src/wp-includes/functions.php

```diff
@@ -7635,3 +7633,85 @@ function is_php_version_compatible( $required ) {
+ * @param bool      $double_quotes  Optional. Uses double quotes `"` to wrap attribute values if true, single quotes `'` otherwise.
+ * @return string  Inline JavaScript code wrapped around <script> tags if $echo is true, null otherwise.
+ */
+function wp_sanitize_script_attributes( $attributes = array() ) {
+	if ( ! array_key_exists( 'type', $attributes ) && ! is_admin() && ! current_theme_supports( 'html5', 'script' ) ) {
+		$attributes['type'] = 'text/javascript';
+	}
+	/**
+	 * Filters script attributes.
+	 *
+	 * @param string $attributes    Associative array representing script attributes.
+	 */
+	$attributes = apply_filters( 'wp_script_attributes', $attributes );
+
+	$attributes_string = '';
+	// Only the attribute name is added to $attributes_string for entries with a boolean value, and that are true.
+	foreach ( $attributes as $attribute_name => $attribute_value ) {
+		if ( is_bool( $attribute_value ) ) {
+			if ( $attribute_value && '' !== $attribute_name ) {
+				$attributes_string .= ' ' . $attribute_name;
+			}
+		} else {
+			$attributes_string .= sprintf( ' %s="%s"', $attribute_name, esc_attr( $attribute_value ) );
+		}
+	}
+
+	return $attributes_string;
+}
+
+/**
+ * Wraps inline JavaScript in `<script>` tags.
+ *
+ * Enables to inject attributes in the `<script>` tag via the `wp_script_attributes` filter.
+ * Automatically injects type attribute if needed.
+ *
+ * @param array     $attributes     `<script>` tag attributes.
+ * @param bool      $double_quotes  Optional. Uses double quotes `"` to wrap attribute values if true, single quotes `'` otherwise.
+ * @param bool      $echo           Optional. Prints to the page if true, returns the value otherwise.
+ * @param string    $content        Optional. Code to be wrapped around `<script>` tags.
+ * @return string|null  Inline JavaScript code wrapped around `<script>` tags if $echo is true, null otherwise.
+ */
+function wp_print_script_loader_tag( $attributes, $echo = true ) {
+	$output = sprintf( "<script%s></script>\n", wp_sanitize_script_attributes( $attributes ) );
+
+	if ( $echo ) {
+		echo $output;
+	} else {
+		return $output;
+	}
+}
+
+/**
+ * Wraps inline JavaScript in `<script>` tags.
+ *
+ * Enables to inject attributes in the `<script>` tag via the `wp_script_attributes` filter.
+ * Automatically injects type attribute if needed.
+ *
+ * @param string    $javascript     Inline JavaScript code.
+ * @param array     $attributes     Optional. `<script>` tag attributes.
+ * @param bool      $double_quotes  Optional. Uses double quotes `"` to wrap attribute values if true, single quotes `'` otherwise.
+ * @param bool      $echo           Optional. Prints to the page if true, returns the value otherwise.
+ * @return string|null  Inline JavaScript code wrapped around `<script>` tags if $echo is true, null otherwise.
+ */
+function wp_print_inline_script_tag( $javascript, $attributes = array(), $echo = true ) {
```

Comment:
When would someone **not** want to `echo` this? The function's name is `wp_print_inline_script_tag`, not `wp_get_inline_script_tag`, so let's remove the `echo` parameter or introduce `wp_get_inline_script_tag` if really needed.

Reply:
When scripts tag need to be appended to another string.
Example: https://github.com/enricocarraro/wordpress-develop/blob/cda80692b51cd2d9705ba8543f4ab36b14518b20/src/wp-includes/class.wp-scripts.php#L294-L308

Reply:
Decoupled the functionality in two functions.